### PR TITLE
feat: useRange match useFacet hook

### PR DESCRIPTION
**Pull request by TomJas-nosto**
## Context

useRange hook should receive same treatment as useFacet:
- Exposing toggleActive function for an active filter
- Exposing active boolean

Refactored utility functions at useRange hook - for easier maintainability and readability

### CODE REVIEW

File: packages/preact/hooks/src/useRange.ts [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
 import { parseNumber } from "@utils/parseNumber"
+import { useCallback, useState } from "preact/hooks"
 
 import { useActions } from "./useActions"
 import { useNostoAppState } from "./useNostoAppState"
 
+type RangeProps = [number | undefined, number | undefined]
+
+const getRangeValues = (filter?: InputSearchTopLevelFilter) => {
+  const filterValue = filter?.range ? filter.range[0] : undefined
```
Comment (timowestnosto) [inline packages/preact/hooks/src/useRange.ts:12]: what's the type of filterValue here?

File: packages/preact/hooks/src/useRange.ts [hunk 2]
```diff
@@ -16,71 +54,73 @@ import { useNostoAppState } from "./useNostoAppState"
...
-  const filterValue = filter?.range ? filter.range[0] : undefined
-  const value =
-    typeof filterValue === "object" && ("gte" in filterValue || "lte" in filterValue)
-      ? ([parseNumber(filterValue.gte), parseNumber(filterValue.lte)] as const)
-      : ([undefined, undefined] as const)
-
-  const min = "min" in stat ? Math.floor(stat.min) : 0
-  const max = "max" in stat ? Math.ceil(stat.max) : 0
-
-  const updateRange = ([from, to]: [number | undefined, number | undefined]) => {
```
[the remark this answers is not in the document]
Comment (TomJas-nosto) [reply packages/preact/hooks/src/useRange.ts:60]: On top of the old ones, I've added some edge cases tests

File: packages/preact/hooks/test/hooks/useRange.spec.ts [hunk 3]
```diff
@@ -100,6 +101,48 @@ describe("useRange", () => {
...
+      query: {
+        products: {
+          filter: [{ field: "price", range: [{ lte: "50" }] }]
+        }
+      }
+    })
+    const render4 = renderHookWithProviders(() => useRange("price"), { store })
+    active = render4.result.current.active
+    expect(active).toBe(true)
+  })
```
Comment (maijalintunokka) [inline packages/preact/hooks/test/hooks/useRange.spec.ts:144]: These are clearly 4 test cases in one, could you split them up? 🙂

File: packages/preact/hooks/src/useRange.ts [hunk 4]
```diff
@@ -1,13 +1,51 @@
...
+  if (fromDefined && fromRounded !== min) {
+    filter.gte = fromRounded.toString()
+  }
+
+  if (toDefined && toRounded !== max) {
+    filter.lte = toRounded.toString()
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
```
Comment (maijalintunokka) [inline packages/preact/hooks/src/useRange.ts:42]: Those are utility functions that are only used here, right? In that case, could you move them below the main body of the hook? JS hoisting allows us to reorder functions in any way we want, so I feel like it makes more sense to have dependencies after the usages. In this case - utility functions after the main hook definition (just below in the file).

File: packages/preact/hooks/src/useRange.ts [hunk 5]
```diff
@@ -16,71 +54,73 @@ import { useNostoAppState } from "./useNostoAppState"
...
+  const hasActiveFilter = value[0] !== undefined || value[1] !== undefined
+  const [active, setActive] = useState(hasActiveFilter)
+
+  const toggleActive = useCallback(() => {
+    setActive(prev => !prev)
+  }, [])
+
+  const updateRange = useCallback(
+    ([from, to]: RangeProps) => {
+      if (!stat) return
```
Comment (maijalintunokka) [inline packages/preact/hooks/src/useRange.ts:98]: I see ChatGPT's code.

Our convention is to never omit the braces.
```
if (!stat) {
  return
}
```
